lib/OpenQA/Worker.pm

```diff
@@ -377,8 +377,21 @@ sub configure_cache_client {
     #       would appear to be stuck and not even respond to signals.
     return delete $self->{_cache_service_client} unless $self->settings->global_settings->{CACHEDIRECTORY};
     my $client = $self->{_cache_service_client} = OpenQA::CacheService::Client->new;
+    my $ua     = $client->ua;
     $client->attempts(1);
-    $client->ua->inactivity_timeout($ENV{OPENQA_WORKER_CACHE_SERVICE_CHECK_INACTIVITY_TIMEOUT} // 10);
+    $ua->inactivity_timeout($ENV{OPENQA_WORKER_CACHE_SERVICE_CHECK_INACTIVITY_TIMEOUT} // 10);
+
+    # set PeerAddrInfo directly to workaround getaddrinfo() being stuck in error state "Address family for hostname
+    # not supported" for local connections (see https://progress.opensuse.org/issues/78390#note-38)
+    # note: The socket_options function has only been added in Mojolicious 8.72. For older versions we still rely
+    #       on the monkey_patch within the BEGIN block. It could be removed when we stop supporting older Mojolicious
+    #       versions.
+    my %cache_service_address = (
+        family   => AF_INET,
+        protocol => IPPROTO_TCP,
+        socktype => SOCK_STREAM,
+        addr     => pack_sockaddr_in('9530', inet_aton('127.0.0.1')));
+    $ua->socket_options({PeerAddrInfo => [\%cache_service_address]}) if $ua->can('socket_options');
```

Comment:
Better use `$ua->socket_options->{PeerAddrInfo} = [\%cache_service_address]`. Then you don't end up accidentally overriding other options that might have been set somewhere else (perhaps even inside `Mojo::UserAgent`).

Reply:
Right. I also moved it to the client module itself to catch all occurrences.